mbuild/formats/lammpsdata.py

```diff
@@ -856,29 +550,85 @@ def write_lammpsdata(
 
+    return
+
+
+def _evaluate_lj_conversion_factors(
+    structure, conversion_name, conversion_factor
+):
+    """Get Lennard Jones style conversion factors. `conversion_name`` can be sigma, epsilon, or mass."""
+    if conversion_factor is None:
+        # Check if structure is parametrized
+        if any([atom.sigma for atom in structure.atoms]) is None:
+            raise ValueError(
+                "LJ units specified but one or more atoms has undefined LJ "
+                "parameters."
+            )
+        else:
+            conversion_factor = np.max(
+                [getattr(atom, conversion_name) for atom in structure.atoms]
+            )
+            warn(
+                f"Assuming {conversion_name} conversion factor of "
+                + str(conversion_factor)
+            )
+        if conversion_factor == 0:
+            conversion_factor = 1
+            warn(
+                f"{conversion_name} conversion factor cannot be inferred from the maximum {conversion_name} value in the ParmEd Structure. "
+                "Setting the {conversion_name} conversion factor to 1"
+            )
+    elif conversion_factor <= 0:
+        raise ValueError(
+            f"The {conversion_name} conversion factor to convert to LJ units should be greater than 0."
+        )
+    else:
+        # assume conversion factor passed in mbuild units, convert to lammps real units
+        if conversion_name == "sigma":
+            conversion_factor *= NM_TO_ANG
+        elif conversion_name == "epsilon":
+            conversion_factor *= 1 / KCAL_TO_KJ
+        elif conversion_name == "mass":
+            pass
+    return conversion_factor
+
+
+def _check_minsmaxs(mins, maxs):
+    """Return True if both mins and maxs have been defined, and each have length 3 otherwise returns False."""
+    if mins and maxs:
+        if len(mins) == 3 and len(maxs) == 3:
+            return True
+        else:
+            warn(
+                "mins and maxs passed to write_lammpsdata, but list size is "
+                "incorrect. mins and maxs will be ignored."
+            )
+            return False
+    else:
+        return False
+
 
 def _get_bond_types(
     structure, bonds, sigma_conversion_factor, epsilon_conversion_factor
 ):
-    unique_bond_types = dict(
+    """Will get the bond types from a parmed structure and convert them to lammps real units."""
+    unique_bond_types = OrderedDict(
         enumerate(
-            set(
-                [
-                    (
-                        round(
-                            bond.type.k
-                            * (
-                                sigma_conversion_factor ** 2
-                                / epsilon_conversion_factor
-                            ),
-                            3,
+            [
+                (
+                    round(
+                        bond.type.k
+                        * (
+                            sigma_conversion_factor ** 2
```

Comment:
If I recall correctly, there were issues where Parmed had assumed (1/2)*k for harmonic bond force constants and I had to reverse that change by including a factor of 2 for all bond constants. Is this accounted for?

Reply:
It is accounted for! I think Parmed grabs the foyer 1/2K values and converts them to K values internally. But from manually checking, I get the parameter I would guess. You might want to look at those unit tests closely though to be 100% certain, because you're right there's no conversion when grabbing bond.type. which is a tad concerning. But my manual checks make sense to me and don't break the real units style which we know was working before.